src/ConnectedKubernetes/ConnectedKubernetes.Autorest/custom/Set-AzConnectedKubernetes.ps1

```diff
@@ -0,0 +1,726 @@
+            return
+        }
+        if (Test-Path Env:HELMREGISTRY) {
+            $RegistryPath = Get-ChildItem -Path Env:HELMREGISTRY
+        } else {
+            $ReleaseTrain = ''
+            if ((Test-Path Env:RELEASETRAIN) -and (Test-Path Env:RELEASETRAIN)) {
+                $ReleaseTrain = Get-ChildItem -Path Env:RELEASETRAIN
+            } else {
+                $ReleaseTrain = 'stable'
+            }
+            $AzLocation = Get-AzLocation | Where-Object { ($_.DisplayName -ieq $Location) -or ($_.Location -ieq $Location)}
+            $Region = $AzLocation.Location
+            if ($null -eq $Region) {
+                Write-Error "Invalid location: $Location"
+                return
+            } else {
+                $Location = $Region
+            }
+            $ChartLocationUrl = "https://${Location}.dp.kubernetesconfiguration.azure.com/azure-arc-k8sagents/GetLatestHelmPackagePath?api-version=2019-11-01-preview&releaseTrain=${ReleaseTrain}"
+
+            $Uri = [System.Uri]::New($ChartLocationUrl)
+            $Account = [Microsoft.Azure.Commands.Common.Authentication.Abstractions.AzureRmProfileProvider]::Instance.Profile.DefaultContext.Account
+            $Env = [Microsoft.Azure.Commands.Common.Authentication.Abstractions.AzureEnvironment]::PublicEnvironments[[Microsoft.Azure.Commands.Common.Authentication.Abstractions.EnvironmentName]::AzureCloud]
+            $TenantId = [Microsoft.Azure.Commands.Common.Authentication.Abstractions.AzureRmProfileProvider]::Instance.Profile.DefaultContext.Tenant.Id
+            $PromptBehavior = [Microsoft.Azure.Commands.Common.Authentication.ShowDialog]::Never
+            $Token = [Microsoft.Azure.Commands.Common.Authentication.AzureSession]::Instance.AuthenticationFactory.Authenticate($account, $env, $tenantId, $null, $promptBehavior, $null)
+            $AccessToken = $Token.AccessToken
+
+            $HeaderParameter = @{
+                "Authorization" = "Bearer $AccessToken"
+            }
+
+            $Response = Invoke-WebRequest -Uri $Uri -Headers $HeaderParameter -Method Post -UseBasicParsing
+            if ($Response.StatusCode -eq 200) {
+                $RegistryPath = ($Response.Content | ConvertFrom-Json).repositoryPath
+            } else {
+                Write-Error "Error while fetching helm chart registry path: ${$Response.RawContent}"
+                return
+            }
+        }
+        Set-Item -Path Env:HELM_EXPERIMENTAL_OCI -Value 1
+
+        # $HelmChartPath = Join-Path -Path $ChartExportPath -ChildPath 'azure-arc-k8sagents'
+        # if (Test-Path Env:HELMCHART) {
+        #     $ChartPath = Get-ChildItem -Path Env:HELMCHART
+        # } else {
+        #     $ChartPath = $HelmChartPath
+        # }
+
+        #Region helm options
+        # !!PDS: The az cli also sets the "proxy" fields in the settings and
```

Comment:
Do check the latest changes to `New-...` and merging them into here.

Reply:
Yes, I have the changes from following commits:
- https://github.com/xw-zhang24/azure-powershell/commit/d822f7a4f50339e3838f38dc4b346761c18e6ebe
-  https://github.com/xw-zhang24/azure-powershell/commit/f57684b7ab95d1a5b236b1b995f62659a0b36ae6